metadata-ingestion/src/datahub/ingestion/source/state_provider/file_ingestion_checkpointing_provider.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import pathlib
+from datetime import datetime
+from typing import Any, Dict, List, Optional, cast
+
+from datahub.configuration.common import ConfigurationError
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.ingestion_job_checkpointing_provider_base import (
+    IngestionCheckpointingProviderBase,
+    IngestionCheckpointingProviderConfig,
+    JobId,
+)
+from datahub.ingestion.sink.file import write_metadata_file
+from datahub.ingestion.source.file import read_metadata_file
+from datahub.metadata.schema_classes import DatahubIngestionCheckpointClass
+
+logger = logging.getLogger(__name__)
+
+
+class FileIngestionStateProviderConfig(IngestionCheckpointingProviderConfig):
+    filename: Optional[str] = None
+
+
+class FileIngestionCheckpointingProvider(IngestionCheckpointingProviderBase):
+    orchestrator_name: str = "file"
+
+    def __init__(self, filename: str, name: str):
+        super().__init__(name)
+        self.filename = filename
+        self.committed = False
+
+    @classmethod
+    def create(
+        cls, config_dict: Dict[str, Any], ctx: PipelineContext, name: str
+    ) -> "FileIngestionCheckpointingProvider":
+        if config_dict is None:
+            raise ConfigurationError("Missing provider configuration.")
+        else:
+            provider_config = FileIngestionStateProviderConfig.parse_obj_allow_extras(
+                config_dict
+            )
+            if provider_config.filename:
+                return cls(provider_config.filename, name)
+            else:
+                raise ConfigurationError(
```

Comment:
the validation logic should go in the pydantic FileIngestionStateProviderConfig class

the create method should be pretty simple - look at other `def create` methods in our codebase for reference